luda-adventure/src/app/game/game.rs

```diff
@@ -0,0 +1,128 @@
+use super::{player_character::new_player, *};
+use namui::prelude::*;
+
+const NEAR_FUTURE: Time = Time::Sec(5.0);
+
+pub struct Game {
+    pub state: GameState,
+    pub camera: Camera,
+    pub ecs_app: crate::ecs::App,
+}
+impl Game {
+    pub fn new() -> Self {
+        let character = mock_character();
+        let quest_object = mock_quest_object();
+        let walls = mock_walls();
+        let floors = mock_floor();
+        let state = GameState::new();
+
+        let player_entity_id = character.id();
+
+        let mut ecs_app = crate::ecs::App::new();
+        ecs_app.add_entity(character);
+        ecs_app.add_entity(quest_object);
+        ecs_app.add_entities(walls);
+        ecs_app.add_entity(floors);
+
+        Self {
+            state,
+            camera: Camera::new(Some(CameraSubject::Object {
+                id: player_entity_id,
+            })),
+            ecs_app,
+        }
+    }
+
+    pub fn update(&mut self, event: &dyn std::any::Any) {
+        let current_time = now();
+        if let Some(event) = event.downcast_ref::<NamuiEvent>() {
+            match event {
+                NamuiEvent::KeyDown(_) => {
+                    self.handle_user_input_for_character_movement(current_time)
+                }
+                NamuiEvent::KeyUp(_) => self.handle_user_input_for_character_movement(current_time),
+                NamuiEvent::AnimationFrame
+                | NamuiEvent::MouseDown(_)
+                | NamuiEvent::MouseUp(_)
+                | NamuiEvent::MouseMove(_)
+                | NamuiEvent::ScreenResize(_)
+                | NamuiEvent::Wheel(_)
+                | NamuiEvent::DeepLinkOpened(_) => (),
+            }
+        }
+
+        while need_to_calculate_next_motion_of_character(&self.ecs_app, current_time, NEAR_FUTURE) {
+            calculate_next_movement_of_character(&mut self.ecs_app, NEAR_FUTURE);
+        }
```

Comment:
I think game don't have to check whether character needs to calculate next motion or not. it's not the responsibility for game.

This also makes performance poor because it query on every loop for each `need_to_calculate_next_motion_of_character` and `calculate_next_movement_of_character`, including collision merging.

What about letting them do this inside?

Reply:
Isn't it easier to read? If not, who do you think should be responsible?

I don't think we need to seriously consider optimization now that we're making prototypes.

Reply:
`game` just activate the function, other jobs will be done inside of job.
`game` should not control inside of the job, before they need to make a branches of logic.
the function you will call need to have the responsibility and it should hide the implementation from outside by [Information_hiding](https://en.wikipedia.org/wiki/Information_hiding).

Reply:
I intended to show the flow. I will hide this.

Reply:
It's enough to show how important this flow is important with function name or comment.